{"name": "Contact form", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"(//td[@class ='key' and contains(text(),'Name')]/following-sibling::td[@class ='value breakword'])[1]\"}\n  (Session info: chrome-headless-shell=121.0.6167.139); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n0   chromedriver                        0x000000010135e7dc chromedriver + 4040668\n1   chromedriver                        0x00000001013569e0 chromedriver + 4008416\n2   chromedriver                        0x0000000100fc9870 chromedriver + 284784\n3   chromedriver                        0x000000010100d080 chromedriver + 561280\n4   chromedriver                        0x0000000101047048 chromedriver + 798792\n5   chromedriver                        0x000000010100174c chromedriver + 513868\n6   chromedriver                        0x0000000101002044 chromedriver + 516164\n7   chromedriver                        0x0000000101323a04 chromedriver + 3799556\n8   chromedriver                        0x0000000101327ee4 chromedriver + 3817188\n9   chromedriver                        0x000000010130c260 chromedriver + 3703392\n10  chromedriver                        0x0000000101328a2c chromedriver + 3820076\n11  chromedriver                        0x00000001012ff01c chromedriver + 3649564\n12  chromedriver                        0x0000000101345e3c chromedriver + 3939900\n13  chromedriver                        0x0000000101345fb4 chromedriver + 3940276\n14  chromedriver                        0x0000000101356660 chromedriver + 4007520\n15  libsystem_pthread.dylib             0x000000018292a034 _pthread_start + 136\n16  libsystem_pthread.dylib             0x0000000182924e3c thread_start + 8", "trace": "get_pardot = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"be26e5676269199e243c1d67e79cd53b\")>\n\n    @allure.tag(\"Contact form\")\n    @allure.severity(allure.severity_level.CRITICAL)\n    @allure.title('Contact form')\n    @allure.description('Compare pardot data with entered contact form data')\n    def test_contact_page(get_pardot):\n        driver = get_pardot\n        pardot = PardotContactPage(driver, Contact_pardot.url_contact)\n        driver.set_window_size(1920, 1080)\n>       pardot.compare_data_contact_page(driver)\n\ntest_pardot/test_contact_pardot.py:14: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n../Pages/Pardot/pardot_contact_page.py:11: in compare_data_contact_page\n    contact_page.compare_data(contact_data_main_contact)\n../Pages/Pardot/Base_pardot_contact.py:46: in compare_data\n    name_pardot = self.driver.find_element(By.XPATH, Contact_pardot.name_field).text.strip()\n/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:741: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x103a34750>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"no such element: Unable to locate element: {\\...00018292a034 _pthread_start + 136\\\\n16  libsystem_pthread.dylib             0x0000000182924e3c thread_start + 8\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"(//td[@class ='key' and contains(text(),'Name')]/following-sibling::td[@class ='value breakword'])[1]\"}\nE         (Session info: chrome-headless-shell=121.0.6167.139); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       0   chromedriver                        0x000000010135e7dc chromedriver + 4040668\nE       1   chromedriver                        0x00000001013569e0 chromedriver + 4008416\nE       2   chromedriver                        0x0000000100fc9870 chromedriver + 284784\nE       3   chromedriver                        0x000000010100d080 chromedriver + 561280\nE       4   chromedriver                        0x0000000101047048 chromedriver + 798792\nE       5   chromedriver                        0x000000010100174c chromedriver + 513868\nE       6   chromedriver                        0x0000000101002044 chromedriver + 516164\nE       7   chromedriver                        0x0000000101323a04 chromedriver + 3799556\nE       8   chromedriver                        0x0000000101327ee4 chromedriver + 3817188\nE       9   chromedriver                        0x000000010130c260 chromedriver + 3703392\nE       10  chromedriver                        0x0000000101328a2c chromedriver + 3820076\nE       11  chromedriver                        0x00000001012ff01c chromedriver + 3649564\nE       12  chromedriver                        0x0000000101345e3c chromedriver + 3939900\nE       13  chromedriver                        0x0000000101345fb4 chromedriver + 3940276\nE       14  chromedriver                        0x0000000101356660 chromedriver + 4007520\nE       15  libsystem_pthread.dylib             0x000000018292a034 _pthread_start + 136\nE       16  libsystem_pthread.dylib             0x0000000182924e3c thread_start + 8\n\n/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py:229: NoSuchElementException"}, "description": "Compare pardot data with entered contact form data", "steps": [{"name": "Comparing contact data", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"(//td[@class ='key' and contains(text(),'Name')]/following-sibling::td[@class ='value breakword'])[1]\"}\n  (Session info: chrome-headless-shell=121.0.6167.139); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n0   chromedriver                        0x000000010135e7dc chromedriver + 4040668\n1   chromedriver                        0x00000001013569e0 chromedriver + 4008416\n2   chromedriver                        0x0000000100fc9870 chromedriver + 284784\n3   chromedriver                        0x000000010100d080 chromedriver + 561280\n4   chromedriver                        0x0000000101047048 chromedriver + 798792\n5   chromedriver                        0x000000010100174c chromedriver + 513868\n6   chromedriver                        0x0000000101002044 chromedriver + 516164\n7   chromedriver                        0x0000000101323a04 chromedriver + 3799556\n8   chromedriver                        0x0000000101327ee4 chromedriver + 3817188\n9   chromedriver                        0x000000010130c260 chromedriver + 3703392\n10  chromedriver                        0x0000000101328a2c chromedriver + 3820076\n11  chromedriver                        0x00000001012ff01c chromedriver + 3649564\n12  chromedriver                        0x0000000101345e3c chromedriver + 3939900\n13  chromedriver                        0x0000000101345fb4 chromedriver + 3940276\n14  chromedriver                        0x0000000101356660 chromedriver + 4007520\n15  libsystem_pthread.dylib             0x000000018292a034 _pthread_start + 136\n16  libsystem_pthread.dylib             0x0000000182924e3c thread_start + 8\n\n", "trace": "  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/evgeniapokrovskaya/Documents/Skai_testing/Pages/Pardot/pardot_contact_page.py\", line 11, in compare_data_contact_page\n    contact_page.compare_data(contact_data_main_contact)\n  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/evgeniapokrovskaya/Documents/Skai_testing/Pages/Pardot/Base_pardot_contact.py\", line 46, in compare_data\n    name_pardot = self.driver.find_element(By.XPATH, Contact_pardot.name_field).text.strip()\n                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py\", line 741, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "steps": [{"name": "Loading JSON data", "status": "passed", "parameters": [{"name": "json_path", "value": "'../../Pages/json/entered_data_contact.json'"}], "start": 1707135862467, "stop": 1707135862468}, {"name": "Compare data", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"(//td[@class ='key' and contains(text(),'Name')]/following-sibling::td[@class ='value breakword'])[1]\"}\n  (Session info: chrome-headless-shell=121.0.6167.139); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n0   chromedriver                        0x000000010135e7dc chromedriver + 4040668\n1   chromedriver                        0x00000001013569e0 chromedriver + 4008416\n2   chromedriver                        0x0000000100fc9870 chromedriver + 284784\n3   chromedriver                        0x000000010100d080 chromedriver + 561280\n4   chromedriver                        0x0000000101047048 chromedriver + 798792\n5   chromedriver                        0x000000010100174c chromedriver + 513868\n6   chromedriver                        0x0000000101002044 chromedriver + 516164\n7   chromedriver                        0x0000000101323a04 chromedriver + 3799556\n8   chromedriver                        0x0000000101327ee4 chromedriver + 3817188\n9   chromedriver                        0x000000010130c260 chromedriver + 3703392\n10  chromedriver                        0x0000000101328a2c chromedriver + 3820076\n11  chromedriver                        0x00000001012ff01c chromedriver + 3649564\n12  chromedriver                        0x0000000101345e3c chromedriver + 3939900\n13  chromedriver                        0x0000000101345fb4 chromedriver + 3940276\n14  chromedriver                        0x0000000101356660 chromedriver + 4007520\n15  libsystem_pthread.dylib             0x000000018292a034 _pthread_start + 136\n16  libsystem_pthread.dylib             0x0000000182924e3c thread_start + 8\n\n", "trace": "  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/evgeniapokrovskaya/Documents/Skai_testing/Pages/Pardot/Base_pardot_contact.py\", line 46, in compare_data\n    name_pardot = self.driver.find_element(By.XPATH, Contact_pardot.name_field).text.strip()\n                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py\", line 741, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "steps": [{"name": "Open the page", "status": "passed", "start": 1707135862468, "stop": 1707135864424}, {"name": "Scroll to element", "status": "passed", "parameters": [{"name": "element_locator", "value": "'//span[@class='slds']/a[1]'"}], "start": 1707135865975, "stop": 1707135865987}], "parameters": [{"name": "contact_data", "value": "{'name': 'Internal test', 'email': 'ashleyholland@example.com', 'last_name': 'Quinn', 'company_name': 'Dizzain', 'job_title': 'QA', 'area_of_int': 'App Marketing', 'm_dig_spend': '$1M+ USD per month', 'country': 'United States', 'state': 'New Hampshire', 'text_c': 'Either never technology morning get that southern. Various into provide place important.\\nRoad thing prove trouble east training. Night camera back send. Sport bar science work she if.', 'how_heard': 'Other', 'other': 'Program especially land interest. Idea artist similar ahead say. Factor eight series second west.\\nIndeed western while add scientist and word fast. Ten candidate purpose care.'}"}], "start": 1707135862468, "stop": 1707135869213}], "parameters": [{"name": "driver", "value": "<selenium.webdriver.chrome.webdriver.WebDriver (session=\"be26e5676269199e243c1d67e79cd53b\")>"}], "start": 1707135862467, "stop": 1707135869216}], "start": 1707135860107, "stop": 1707135869217, "uuid": "16d1627e-8214-48eb-b045-1384b3058592", "historyId": "cfd7010345844d018d2463b5999ca585", "testCaseId": "cfd7010345844d018d2463b5999ca585", "fullName": "test_pardot.test_contact_pardot#test_contact_page", "labels": [{"name": "tag", "value": "Contact form"}, {"name": "severity", "value": "critical"}, {"name": "parentSuite", "value": "test_pardot"}, {"name": "suite", "value": "test_contact_pardot"}, {"name": "host", "value": "MacBook-Pro.local"}, {"name": "thread", "value": "18565-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test_pardot.test_contact_pardot"}]}